Macros/e3 Includes/e3_Basics.inc

```diff
@@ -2055,8 +2055,9 @@ SUB check_Supply
   /if (${Waiting4Rez}) /return
   /if (${Bool[${Me.Buff[Resurrection Sickness]}]}) /return
   | Extra checks here because sometimes Waiting4Rez doesn't seem to short out supply
-  /if (!${Bool[${InvSlot[pack8].Item.ID}]}) /return 
-  /if (${Me.MaxHPs} < 10000) /return
+  | Gloves (12) and belt (20)
+  /if (!${Bool[${InvSlot[12].Item.ID}]} && !${Bool[${InvSlot[20].Item.ID}]}) /return
```

Comment:
I would combine these so it would be
If no gloves/belt and your buffs are either 0 or 1, You are dead. ATM they fire off independently.

Reply:
Also probably add this to a method that can be used elsewhere. Probably need a "you are dead" to handle situations for where waitforrez fails. It can become the new waitforrez. I would expand to the entire item slots however and also check for the waitforrez. 
Something like check_IfDead, so would be 
/call check_ifDead
/if (${AmIDead}) /return

Reply:
you can pick a better name :P

Reply:
Also would probably add it to the utilities.inc

Reply:
I think for the purposes of Supply specifically, a minimal fix like I did should *usually* be fine, but I agree that something a bit more thorough would be preferred. I thought about it for a bit and I am not sure theres a perfect solution, but basically we can check their gear, their bagslots and their buffs and if any 2 of those 3 are missing more than some threshold (I went with >half your gear, > half bagslots have no bag for now) its probably safe to assume they are dead. I threw together this macro to test it out, seems to do the thing. I commented some of the concerns I thought of in there in the file. Let me know what you think and I'll add it. [https://github.com/istivaan/eq_macros/blob/main/AmIDead.mac](url)

Reply:
Ok couple things. I made it so it does a check for 1 item in equipment or 1 item in inventory thats not temporary just to account for cases I could think of. However, I started looking at e3_Wait4Rez.inc and theres a problem it only sets ${Waiting4Rez} if its not defined yet and theres a couple places where that happens. So maybe we just need an else { /varset Waiting4Rez etc. Although TBH Its incredibly annoying when supply does stuff when youre waiting to be rezzed, I'd still like to have the sanity check in that function but I think we can fix Wait4Rez and thats good enough for most issues?

Reply:
Also heres my updated AmIDead, pretty much ready to paste into utils except for the main and buff check methods https://github.com/istivaan/eq_macros/blob/main/AmIDead.mac